### PR TITLE
UIPFI-133 Staff suppress facet - default option "No" selected

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Staff suppress facet - default option "No" selected
Similar approach to Inventory - set default `resources.query` properties and track user interaction with staff suppress filter with session storage flag.

## Screenshots

https://github.com/folio-org/ui-plugin-find-instance/assets/19309423/7f8ec6c6-74da-479b-a003-ebeb42b08baf

## Issues
[UIPFI-133](https://folio-org.atlassian.net/browse/UIPFI-133)

### CODE REVIEW

File: Imports/imports/InstanceFilters/InstanceFilters.js [hunk 1]
```diff
@@ -108,6 +109,22 @@ const InstanceFilters = ({
     },
   ];
 
+  const clearStaffSuppressStorageFlag = useCallback(() => {
+    sessionStorage.setItem(USER_TOUCHED_STAFF_SUPPRESS_STORAGE_KEY, true);
```
Comment (mariia-aloshyna) [inline Imports/imports/InstanceFilters/InstanceFilters.js:113]: so clear means that staff suppressed is `true`?

File: InstanceSearch/FindInstanceContainer.js [hunk 2]
```diff
@@ -183,6 +203,15 @@ class FindInstanceContainer extends React.Component {
     this.source.update(this.props);
   }
 
+  componentWillUnmount() {
+    window.removeEventListener('beforeunload', this.handleBeforeUnload);
+    sessionStorage.setItem(USER_TOUCHED_STAFF_SUPPRESS_STORAGE_KEY, false);
```
Comment (mariia-aloshyna) [inline InstanceSearch/FindInstanceContainer.js:208]: can `this.handleBeforeUnload` be used here?

File: Imports/imports/InstanceFilters/InstanceFilters.js (hunk 1 above)
Comment (BogdanDenis) [reply Imports/imports/InstanceFilters/InstanceFilters.js:113]: Thanks, fixed

**Review by mariia-aloshyna — APPROVED** [review]

**Review by Terala-Priyanka — APPROVED** [review]

File: InstanceSearch/FindInstanceContainer.test.js [hunk 3]
```diff
@@ -126,36 +129,90 @@ const renderFindInstanceContainer = (prop) => render(
...
-        },
-        query: {
-          sort: ''
-        }
-      };
-      const logger = 'logger';
-      buildQuery(queryParams, pathComponents, resourceData, logger, defaultProps);
-      expect(makeQueryFunction).toBeCalled();
+
+  describe.only('buildQuery', () => {
```
Comment (Dmytro-Melnyshyn) [inline InstanceSearch/FindInstanceContainer.test.js:152]: only

**Pull request merged**